theme/static/js/hs-vue/geoconnex-reference-collection.js

```diff
@@ -0,0 +1,1216 @@
+      feature.relative_id = feature.properties.uri.split("ref/").pop();
+      feature.collection = feature.relative_id.split("/")[0];
+      feature.uri = feature.properties.uri;
+      geoconnexApp.setFeatureName(feature);
+      feature.text = `${feature.NAME} [${feature.relative_id}]`;
+
+      //prevent duplicate selections
+      geoconnexApp.selectedReferenceItems.forEach((it) => {
+        if (feature.uri === it.value) {
+          feature.disabled = true;
+        }
+      });
+      return feature;
+    },
+    isUrl(stringToTest) {
+      try {
+        new URL(stringToTest);
+      } catch (_) {
+        return false;
+      }
+      return true;
+    },
+    trimString(longString) {
+      return longString.length > geoconnexApp.stringLengthLimit
+        ? `${longString.substring(0, geoconnexApp.stringLengthLimit)}...`
+        : longString;
+    },
+    until(conditionFunction) {
+      geoconnexApp.log(`Waiting for [ ${conditionFunction} ] to resolve...`);
+      const poll = (resolve) => {
+        if (conditionFunction()) {
+          geoconnexApp.log(`Promise for [ ${conditionFunction} ] resolved`);
+          resolve();
+        } else setTimeout((_) => poll(resolve), 400);
+      };
+      return new Promise(poll);
+    },
+    isCacheValid(response) {
+      let geoconnexApp = this;
+      if (!response || !response.ok) return false;
+      if (geoconnexApp.enforceCacheDuration) {
+        var fetched = response.headers.get("fetched-on");
+        if (
+          fetched &&
+          parseFloat(fetched) + geoconnexApp.cacheDuration >
+            new Date().getTime()
+        )
+          return true;
+        geoconnexApp.log("Cached data not valid.");
+        return false;
+      }
+      return true;
+    },
+    setCustomItemRules() {
+      let geoconnexApp = this;
+      geoconnexApp.featureRules = [
+        function (v) {
+          let invalid = [];
+          for (let item of v) {
+            try {
+              url = new URL(item.value);
+            } catch (_) {
+              invalid.push(item.text);
+            }
+          }
+          if (invalid.length === 1) {
+            return `"${invalid}" is not a valid URI. We recommend that your custom feature be linkable`;
+          }
+          if (invalid.length === 2) {
+            return `"${invalid.join(
+              '" and "'
+            )}" are not a valid URIs. We recommend that custom features be linkable.`;
+          }
+          if (invalid.length > 2) {
+            return `"${invalid
+              .join('", "')
+              .replace(
+                /, ([^,]*)$/,
+                " and $1"
+              )}" are not a valid URIs. We recommend that custom features be linkable`;
+          }
+          return true;
+        },
+      ];
+    },
+  },
+  beforeMount() {
+    this.setCustomItemRules();
+  },
+  async mounted() {
+    // TODO: artifacts / bits of text/elements are shown during initial load of page (less than 1 sec)
```

Comment:
@Maurier a few weeks ago you mentioned that you might have a way to prevent the DOM artifacts that show briefly during page load. I haven't been able to figure it out...do I need to use a different Lifecycle hook? Thanks!

Reply:
I'll have to try this out and look into it a little closer. Let me know if it's a good time to do so.